### PR TITLE
Remove a validity pattern from stop time

### PR DESCRIPTION
**Pull request by l-vincent-l**
Also integrates the good comparison on overmidnight trains
Adds and fixes some tests.

### CODE REVIEW

**Comment by antoine-de** [comment]
To keep the history, this PR integrates #321

File: source/routing/raptor.cpp [hunk 1]
```diff
@@ -239,29 +239,29 @@ RAPTOR::compute_all(const std::vector<std::pair<type::idx_t, navitia::time_durat
     clear_and_init(departures, calc_dest, bound, clockwise);
 
     boucleRAPTOR(accessibilite_params, clockwise, disruption_active, false, max_transfers);
-    //auto tmp = makePathes(calc_dest, bound, accessibilite_params, *this, clockwise, disruption_active);
-    //result.insert(result.end(), tmp.begin(), tmp.end());
+//    auto tmp = makePathes(calc_dep, calc_dest, accessibilite_params, *this, clockwise, disruption_active);
```
Comment (antoine-de) [inline source/routing/raptor.cpp:242]: on peut virer ces lignes non ?

File: source/routing/raptor.cpp [hunk 2]
```diff
@@ -239,29 +239,29 @@ RAPTOR::compute_all(const std::vector<std::pair<type::idx_t, navitia::time_durat
...
-    } else {
+    }
 
-        //Second passe : permet d’optimiser les temps de correspondance
-        departures = get_solutions(calc_dep, calc_dest, !clockwise,
-                                   labels,
-                                   accessibilite_params, data, disruption_active);
-        for(auto departure : departures) {
-            clear_and_init({departure}, calc_dep, departure_datetime, !clockwise);
+    //Second passe : permet d’optimiser les temps de correspondance
```
Comment (antoine-de) [inline source/routing/raptor.cpp]: Faut qu'on en profite pour traduire les commentaires :)

File: source/routing/raptor.cpp [hunk 3]
```diff
@@ -239,29 +239,29 @@ RAPTOR::compute_all(const std::vector<std::pair<type::idx_t, navitia::time_durat
...
+        clear_and_init({departure}, calc_dep, departure_datetime, !clockwise);
 
-            boucleRAPTOR(accessibilite_params, !clockwise, disruption_active, true, max_transfers);
+        boucleRAPTOR(accessibilite_params, !clockwise, disruption_active, true, max_transfers);
 
-            if(b_dest.best_now_jpp_idx != type::invalid_idx) {
-                std::vector<Path> temp = makePathes(calc_dest, calc_dep, accessibilite_params, *this, !clockwise, disruption_active);
-                result.insert(result.end(), temp.begin(), temp.end());
-            }
+        if(b_dest.best_now_jpp_idx == type::invalid_idx) {
```
Comment (antoine-de) [inline source/routing/raptor.cpp:260]: Ahhhhh t'es enfin d'accord que c'est plus clair de faire un continue tot :dancers:

File: source/routing/tests/best_stoptime.cpp [hunk 4]
```diff
@@ -999,8 +999,11 @@ BOOST_AUTO_TEST_CASE(passe_minuit_2_vp) {
     {
         DateTime dt_test = DateTimeUtils::set(1, sp2_departure - 101);
         std::tie(st1, dt1) = earliest_stop_time(jpp2, dt_test, *(b.data), false, false);
-        BOOST_CHECK_EQUAL(dt1, 0);
-        BOOST_CHECK(st1 == nullptr);
+        BOOST_CHECK_EQUAL(dt1, DateTimeUtils::set(1, sp2_departure));
```
Comment (antoine-de) [inline source/routing/tests/best_stoptime.cpp:1003]: Ca serait top de caler un commentaire pour chaque test sur ce qu'on veut tester, histoire que les faibles comme moi puisse comprendre rapidement.

File: source/routing/tests/best_stoptime.cpp [hunk 5]
```diff
@@ -1033,37 +1036,197 @@ BOOST_AUTO_TEST_CASE(passe_minuit_2_vp) {
...
+        BOOST_CHECK_EQUAL(dt2, DateTimeUtils::set(1, sp3_arrival));
+        BOOST_CHECK(st1 == nullptr);
+        BOOST_REQUIRE(st2 != nullptr);
+        BOOST_CHECK_EQUAL(st2->departure_time, sp3_arrival);
+        BOOST_CHECK_EQUAL(st2->journey_pattern_point->stop_point->stop_area->name, spa3);
+    }
+}
+
+/**
+ * Cas de passe minuit avec changement d'heure entre l'heure d'arrivée et de départ
```
Comment (antoine-de) [inline source/routing/tests/best_stoptime.cpp]: Faut traduire

File: source/type/data.cpp [hunk 6]
```diff
@@ -188,7 +188,7 @@ ValidityPattern* Data::get_similar_validity_pattern(ValidityPattern* vp) const{
 
 ValidityPattern* Data::get_or_create_validity_pattern(ValidityPattern* ref_validity_pattern, const uint32_t time){
 
-    if(time > 24*3600) {
+    if(time >= 24*3600) { //Seconds in a day
```
Comment (antoine-de) [inline source/type/data.cpp]: Il faut supprimer la méthode non ? elle ne doit plus etre utilisée.

**Comment by Tristramg** [comment]
Make some benchmark in memory consumption before/after.
32 bytes per stoptime, that makes almost 500Mbytes for a large set as ÎdF or NL.
Is the comment really used? Can’t it be also thrown away?

Anyway :+1: :heart: :heart_decoration: :heart_eyes: :heart_eyes_cat:

File: source/routing/raptor.cpp (hunk 1 above)
Comment (l-vincent-l) [inline source/routing/raptor.cpp:242]: Nop, this is used for debbugging purposes. I think it's the third time you ask me about it.

**Comment by antoine-de** [comment]
Yep Vincent is doing lot's of bench for memory consumption and perf (but it doesn't seems to have that much effect, strange).

For the comment it seems to be used, but only for display and quite rarely. I think we can thus add a map<stoptime, comment> in data to limit the impact.

File: source/type/data.cpp (hunk 6 above)
Comment (l-vincent-l) [inline source/type/data.cpp]: This is used in : ed/adapted.cpp:42

Comment (antoine-de) [inline source/type/data.cpp]: ok, thus it is better to put Datetime::SECONDS_PER_DAY

**Pull request merged**